piecemeal-plugin/src/main/kotlin/dev/bnorm/piecemeal/plugin/ir/PiecemealBuilderGenerator.kt

```diff
@@ -195,6 +202,62 @@ class PiecemealBuilderGenerator(
     }
   }
 
+  /**
+   * ```kotlin
+   * fun copy(transform: Person.Builder.() -> Unit): Person {
+   *     val tmp = newBuilder()
+   *     tmp.transform()
+   *     return tmp.build()
+   * }
+   * ```
+   */
+  private fun generateCopyFunction(function: IrSimpleFunction): IrBody? {
+    val transformLambda = function.valueParameters.single()
+    val builderType = (transformLambda.type as IrSimpleType).arguments[0] as IrType
+    val builderClass = builderType.classifierOrNull?.owner as? IrClass ?: return null
+
+    val piecemealClass = function.parentAsClass
+    val newBuilderFunction = piecemealClass.functions.single {
+      it.name == NEW_BUILDER_FUN_NAME &&
+        it.extensionReceiverParameter == null &&
+        it.valueParameters.isEmpty()
+    }
+
+    val invoke = transformLambda.type.classOrFail.functions
+      .single {
+        val owner = it.owner
+        owner.name == Name.identifier("invoke") &&
+          owner.extensionReceiverParameter == null &&
+          owner.valueParameters.size == 1 &&
+          owner.valueParameters.single().type.classifierOrNull is IrTypeParameterSymbol
+      }
+
+    val build = builderClass.functions
+      .single {
+        it.name == BUILD_FUN_NAME &&
+          it.extensionReceiverParameter == null &&
+          it.valueParameters.isEmpty()
+      }
```

Comment:
I used name + parameters to more strongly guarantee finding the correct functions here, although I'm not sure the type parameter on `invoke` technically guarantees a unique function. Happy to follow up with a copy/paste to the other TODOs you have if you like this approach.

Reply:
For functions that we generate, I have half an idea to use different `GeneratedDeclarationKey`s for each unique function that is generated. Or maybe add an enum property to the one I already use. This should result in different origins and would allow more exact search and transformation. But this is good for now. 

Leave the TODOs as they are. I want to search the Kotlin repository for any utilities I may have missed.